agent/agents/mysql/slowlog/parser/parser.go

```diff
@@ -167,90 +168,121 @@ func (p *SlowLogParser) Run() {
 func (p *SlowLogParser) parseHeader(line string) {
 	p.logf("header")
 
-	if !headerRe.MatchString(line) {
+	if !isHeader(line) {
 		p.inHeader = false
 		p.inQuery = true
 		p.parseQuery(line)
 		return
 	}
+	line = skipPrefix(line)
 
 	if p.headerLines == 0 {
 		p.event.Offset = p.lineOffset
 	}
 	p.headerLines++
 
 	switch {
-	case strings.HasPrefix(line, "# Time"):
-		p.logf("time")
-		m := timeRe.FindStringSubmatch(line)
+	case strings.HasPrefix(line, "Time: "):
+		p.parseTime(line)
+
+	case strings.HasPrefix(line, "User"):
+		p.parseUser(line)
+
+	case strings.HasPrefix(line, "admin"):
+		p.parseAdmin(line)
+
+	default:
+		p.parseMetrics(line)
+	}
+}
+
+func skipPrefix(line string) string {
+	line = line[2:]
+	return line
+}
+
+func (p *SlowLogParser) parseTime(line string) {
+	p.logf("time")
+	m := timeRe.FindStringSubmatch(line)
+	if len(m) == 2 {
+		p.event.Ts, _ = time.ParseInLocation("060102 15:04:05", m[1], p.opts.DefaultLocation)
+	} else {
+		m = timeNewRe.FindStringSubmatch(line)
 		if len(m) == 2 {
-			p.event.Ts, _ = time.ParseInLocation("060102 15:04:05", m[1], p.opts.DefaultLocation)
+			p.event.Ts, _ = time.ParseInLocation(time.RFC3339Nano, m[1], p.opts.DefaultLocation)
```

Comment:
"060102 15:04:05" -> time.RFC3339Nano

Was it intended? Why we need this change?

Reply:
Maybe it was affected via my editor (looks like replaced automatically)

Reply:
will you revert it back? or are you planning to leave it here?

Reply:
imho if it wasnt intented it should be reverted

Reply:
I got it, you moved this code to another function and now it compares not same lines.
sorry but quality of code is bad and it's not easy to find that.

Reply:
@dliakhov this is not my code originally